### ISSUE TITLE
Build error if you try to use Light fm data sets

### ISSUE THREAD

**Issue by NSAryan12** [lyst/lightfm#134]
if type in python 
```data = fetch_movielens(min_rating=4.0)```

I get an error like this
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/Aryan/anaconda/lib/python2.7/site-packages/lightfm/datasets/movielens.py", line 169, in fetch_movielens
    item_metadata_raw, genres_raw) = _read_raw_data(zip_path)
  File "/Users/Aryan/anaconda/lib/python2.7/site-packages/lightfm/datasets/movielens.py", line 20, in _read_raw_data
    with zipfile.ZipFile(path) as datafile:
  File "/Users/Aryan/anaconda/lib/python2.7/zipfile.py", line 770, in __init__
    self._RealGetContents()
  File "/Users/Aryan/anaconda/lib/python2.7/zipfile.py", line 811, in _RealGetContents
    raise BadZipfile, "File is not a zip file"
zipfile.BadZipfile: File is not a zip file
```

I have imported these lines above

```from lightfm import LightFM```
```from lightfm.datasets import fetch_movielens```

**Comment by maciejkula** [comment]
Can you:
1. Check where it's looking for the data (the `zip_path` above)?
2. Check that the file at that path is a valid zip file?
3. Try deleting that file and it's directory and run the command again.

My first suspicion is that there might have been a problem with the download and you have a partially completed file cached in the data directory.

**Comment by maciejkula** [comment]
Any luck?

**Comment by maciejkula** [comment]
?

**Comment by anshulg8** [comment]
You need to update zip_path in lightfm/datasets/movielens.py

**Comment by maciejkula** [comment]
Are you saying that the code doesn't work? Could you post the trace for the error you encountered?

**Comment by anshulg8** [comment]
I was stuck in a similar issue, which I posted [here](https://stackoverflow.com/questions/46138406/build-error-if-you-try-to-use-lightfm-data-sets).
All I had do was replace this:
```
zip_path = _common.get_data(data_home,
                            ('https://github.com/maciejkula/'
                             'lightfm_datasets/releases/'
                             'download/v0.1.0/movielens.zip'),
                            'movielens100k',
                            'movielens.zip',
                            download_if_missing)
```
with 
```
zip_path = _common.get_data(data_home,
                            ('https://github.com/maciejkula/
                            lightfm_datasets/releases/
                            download/v0.1.0/movielens.zip'),
                            'movielens100k',
                            'movielens.zip',
                            download_if_missing)
```

**Comment by maciejkula** [comment]
Can you point out the difference between the two? I don't quite see it!

**Comment by anshulg8** [comment]
The 4 extra apostrophes(') in the URL were creating problem for me.

**Comment by maciejkula** [comment]
Are you use? Python uses what's called implicit string concatenation; these quotes are simply dropped in the final string. Your changed code is syntactically identical.

**Comment by anshulg8** [comment]
It didn't work for me, I am using Python3.6

**Comment by ParikshitSinghTomar** [comment]
Thanks @anshulg8, for pointing url. While my problem not solved by the help of above solution. What I tried written below.
1. Copy path and download zip file
2. Replace below code
data = fetch_movielens(min_rating=4.0)

with 
def _read_raw_data(path):
       with zipfile.ZipFile(path) as datafile:
        return (datafile.read('ml-100k/ua.base').decode().split('\n'),
                datafile.read('ml-100k/ua.test').decode().split('\n'),
                datafile.read('ml-100k/u.item').decode(errors='ignore').split('\n'),
                datafile.read('ml-100k/u.genre').decode(errors='ignore').split('\n'))

data=_read_raw_data('movielens.zip')
I know this is not right way but I tried it and it works perfect.

**Comment by maciejkula** [comment]
This is not a bug in the code. Your download was corrupted for some reason. Delete the downloaded file and the code should work again

**Comment by maciejkula** [comment]
This problem is solved in the newest release, 1.14.
